dspace-server-webapp/src/test/java/org/dspace/app/rest/builder/CollectionBuilder.java

```diff
@@ -164,58 +165,54 @@ public Collection build() {
 
     @Override
     public void cleanup() throws Exception {
-        deleteAdminGroup();
-        deleteDefaultReadGroups(collection);
-        deleteWorkflowGroups(collection);
-        delete(collection);
-    }
-
-    private void deleteAdminGroup() throws SQLException, AuthorizeException {
-        if (collection.getAdministrators() != null) {
-            try (Context c = new Context()) {
-                c.turnOffAuthorisationSystem();
-                collectionService.removeAdministrators(c, collection);
+       try (Context c = new Context()) {
+            c.turnOffAuthorisationSystem();
+            collection = c.reloadEntity(collection);
```

Comment:
This is a comment for other reviewers to explain the reasoning behind this change (and the same in all the other builder).
It is better to reload the primary object as this will guarantee us to access also fresh **related objects** during the cleanup, in this case the Admin Group.

Before, all the related objects where reloaded just before to be deleted (see previous implementeation of the deleteWorkflowGroups) but this approach have several issues:
1. it is easy to forget about it, indeed when the deleteAdminGroup method was added the first time the build was failing due to that
2. doing that manually each time duplicate lot of code
3. as you only reload the object that was attached to the "old" copy of the primary object you will fail to find related object created in outside thread. If the admin group has been created via a REST call collection.getAdministrators() run before the refresh of the community object would have returned null

Reply:
Just a minor comment, I wonder if we should add a small inline comment here to *remind ourselves* why the `reloadEntity` is necessary. That way we don't accidentally refactor it out at a later time.

We could just say:
`// Ensure object and any related objects are reloaded before checking to see what needs cleanup`